telcell/cell_identity.py

```diff
@@ -0,0 +1,295 @@
+import re
+from typing import Optional, Any
+
+RADIO_GSM = "GSM"
+RADIO_UMTS = "UMTS"
+RADIO_LTE = "LTE"
+RADIO_NR = "NR"
+
+CELL_IDENTITY_PATTERN = re.compile(
+    r'^((?P<radio>[a-zA-Z]+)/)?(?P<mcc>[0-9]+|\?)-(?P<mnc>[0-9]+|\?)'
+    + r'(-((?P<lac>[0-9]+|\?)-(?P<ci>[0-9]+|\?)|(?P<eci>[0-9]+|\?)))?$')
+
+
+class CellIdentity:
+    """
+    An antenna is identified by a Cell Global Identity (CGI) or an E-UTRAN Cell
+    Global Identity (eCGI), both 15 digit codes.
+
+    Resouces:
+        * https://www.cellmapper.net/
+    """
+
+    @staticmethod
+    def create(
+        radio: Optional[str] = None,
+        mcc: Optional[int] = None,
+        mnc: Optional[int] = None,
+        lac: Optional[int] = None,
+        ci: Optional[int] = None,
+        eci: Optional[int] = None,
+    ) -> "CellIdentity":
```

Comment:
Considering they're all optional and the test code passes all of them by name, would it make sense to force them to be keyword arguments?
```suggestion
    def create(*,
        radio: Optional[str] = None,
        mcc: Optional[int] = None,
        mnc: Optional[int] = None,
        lac: Optional[int] = None,
        ci: Optional[int] = None,
        eci: Optional[int] = None,
    ) -> CellIdentity:
```

Also, typing expression are evaluated later, the return type can be unquoted.